### PR TITLE
feat(fiatconnect): Include network fee estimation on FiatConnect Review Screen

### PR DESCRIPTION
**Pull request by jophish**
### Description

For ACT-586. Adds in network fee estimation to the FiatConnect Review screen for cash outs.

### Images

Below is an example of the diff from current behavior, for a cash-out of 0.40 cEUR, with a provider fee of 0.02 cEUR, and an estimated network fee of .047 cEUR. (cEUR was chosen as an illustrative example since it requires conversion of the cUSD-based fee estimation to the user-selected crypto type.) Note that the displayed exchange rate is unchanged, which is, I believe, the desired behavior as suggested in the ticket (since network fee doesn't contribute to the portion of total funds used to actually convert to the desired fiat currency).

2023-01-04-120425_918x512_scrot

2023-01-04-120437_909x484_scrot

### Test plan

Manual and unit tested.

### Related issues

- Fixes ACT-586

### Backwards compatibility

Yes.

### CODE REVIEW

File: src/fiatconnect/ReviewScreen.tsx [hunk 1]
```diff
@@ -339,15 +368,28 @@ function getDisplayAmounts({
...
+          exchangeRates[cryptoType]
+        ) || networkFee
+    }
+
+    const receive = Number(normalizedQuote.getFiatAmount())
+    const total = Number(
+      new BigNumber(normalizedQuote.getCryptoAmount()).plus(networkFee).toString()
+    )
+    const totalFee = Number(providerFee.plus(networkFee))
+    const totalMinusProviderFee = total - totalFee
```
Comment (anthonydaoud) [inline src/fiatconnect/ReviewScreen.tsx:391]: nit: I think this should be named `totalMinusFees` because it includes both the provider and network fees

**Comment by anthonydaoud** [comment]
Looks good to me!

**Review by anthonydaoud — APPROVED** [review]

File: src/fiatconnect/ReviewScreen.tsx [hunk 2]
```diff
@@ -339,15 +368,28 @@ function getDisplayAmounts({
...
-    const totalMinusFee = Number(total) - Number(fee || 0)
-    const exchangeRate = Number(receive) / totalMinusFee
+    const providerFee = normalizedQuote.getFeeInCrypto(exchangeRates) || new BigNumber(0)
+    let networkFee = new BigNumber(0)
+    if (feeEstimate?.usdFee) {
+      networkFee =
+        convertLocalAmountToCurrency(
+          convertDollarsToLocalAmount(new BigNumber(feeEstimate.usdFee), exchangeRates.cUSD),
+          exchangeRates[cryptoType]
+        ) || networkFee
```
[the remark this answers is not in the document]
Comment (jophish) [reply src/fiatconnect/ReviewScreen.tsx:383]: When it returns a falsy result, we fall back on using 0 as an estimation for the network fee -- and yeah, changed to `??`.

File: src/fiatconnect/ReviewScreen.tsx [hunk 3]
```diff
@@ -56,6 +64,19 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
...
+  const cryptoType = normalizedQuote.getCryptoTypeString()
+  const tokenAddress = tokenList.find((token) => token.symbol === cryptoType)?.address
+  const feeEstimates = useSelector(feeEstimatesSelector)
+  const feeEstimate = tokenAddress ? feeEstimates[tokenAddress]?.[feeType] : undefined
+
+  useEffect(() => {
+    if (!feeEstimate && tokenAddress) {
+      dispatch(estimateFee({ feeType, tokenAddress }))
+    }
+  }, [feeEstimate])
```
Comment (cajubelt) [inline src/fiatconnect/ReviewScreen.tsx:78]:
```suggestion
  }, [feeEstimate, tokenAddress])
```

File: src/fiatconnect/ReviewScreen.tsx [hunk 4]
```diff
@@ -339,15 +368,28 @@ function getDisplayAmounts({
...
 }) {
   const cryptoType = normalizedQuote.getCryptoType()
   const fiatType = normalizedQuote.getFiatType()
   if (flow === CICOFlow.CashOut) {
-    const receive = normalizedQuote.getFiatAmount()
-    const total = normalizedQuote.getCryptoAmount()
-    const fee = normalizedQuote.getFeeInCrypto(exchangeRates)
-    const totalMinusFee = Number(total) - Number(fee || 0)
-    const exchangeRate = Number(receive) / totalMinusFee
+    const providerFee = normalizedQuote.getFeeInCrypto(exchangeRates) || new BigNumber(0)
```
Comment (cajubelt) [inline src/fiatconnect/ReviewScreen.tsx:376]:
```suggestion
    const providerFee = normalizedQuote.getFeeInCrypto(exchangeRates) ?? new BigNumber(0)
```

File: src/fiatconnect/ReviewScreen.tsx (hunk 1 above)
Comment (cajubelt) [reply src/fiatconnect/ReviewScreen.tsx:391]: why was this marked resolved? looks like the name is still totalMinusProviderFee

**Review by cajubelt — APPROVED** [review]

File: src/fiatconnect/ReviewScreen.tsx (hunk 1 above)
Comment (jophish) [reply src/fiatconnect/ReviewScreen.tsx:391]: :facepalm: Thanks for the catch... I made the changes and then stashed them to debug something but never re-applied it...

**Pull request merged**